File: CS/TreeListCheckBoxes/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraTreeList.Nodes;
using DevExpress.XtraTreeList;
using DevExpress.XtraTreeList.ViewInfo;

namespace DXSample
{
    public partial class MainForm : XtraForm
    {
        public MainForm()
        {
            InitializeComponent();
            treeList.ExpandAll();
        }

        Rectangle hotTrackCheckBoxRectangle = Rectangle.Empty;

        public Rectangle HotTrackCheckBoxRectangle {
            get { return hotTrackCheckBoxRectangle; }
            set {
                if(hotTrackCheckBoxRectangle == value) return;
                Rectangle prevHotTrackCheckBoxRectangle = hotTrackCheckBoxRectangle;
                hotTrackCheckBoxRectangle = value;
                treeList.Invalidate(prevHotTrackCheckBoxRectangle);
                treeList.Invalidate(hotTrackCheckBoxRectangle);
            }
        }

        private void OnCustomDrawNodeCheckBox(object sender, DevExpress.XtraTreeList.CustomDrawNodeCheckBoxEventArgs e) {
            if(e.Bounds.Equals(HotTrackCheckBoxRectangle))
                e.ObjectArgs.State = DevExpress.Utils.Drawing.ObjectState.Hot;
        }

        private void OnMouseMove(object sender, MouseEventArgs e) {
            TreeList tree = sender as TreeList;
            TreeListHitInfo info = tree.CalcHitInfo(e.Location);
            if (info.HitInfoType == HitInfoType.NodeCheckBox)
                HotTrackCheckBoxRectangle = info.Bounds;
            else
                HotTrackCheckBoxRectangle = Rectangle.Empty;
           }
        }
}
    

File: VB/TreeListCheckBoxes/MainForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraTreeList.Nodes
Imports DevExpress.XtraTreeList
Imports DevExpress.XtraTreeList.ViewInfo

Namespace DXSample
	Partial Public Class MainForm
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
			treeList.ExpandAll()
		End Sub

		Private hotTrackCheckBoxRectangle_Renamed As Rectangle = Rectangle.Empty

		Public Property HotTrackCheckBoxRectangle() As Rectangle
			Get
				Return hotTrackCheckBoxRectangle_Renamed
			End Get
			Set(ByVal value As Rectangle)
				If hotTrackCheckBoxRectangle_Renamed = value Then
					Return
				End If
				Dim prevHotTrackCheckBoxRectangle As Rectangle = hotTrackCheckBoxRectangle_Renamed
				hotTrackCheckBoxRectangle_Renamed = value
				treeList.Invalidate(prevHotTrackCheckBoxRectangle)
				treeList.Invalidate(hotTrackCheckBoxRectangle_Renamed)
			End Set
		End Property

		Private Sub OnCustomDrawNodeCheckBox(ByVal sender As Object, ByVal e As DevExpress.XtraTreeList.CustomDrawNodeCheckBoxEventArgs) Handles treeList.CustomDrawNodeCheckBox
			If e.Bounds.Equals(HotTrackCheckBoxRectangle) Then
				e.ObjectArgs.State = DevExpress.Utils.Drawing.ObjectState.Hot
			End If
		End Sub

		Private Sub OnMouseMove(ByVal sender As Object, ByVal e As MouseEventArgs) Handles treeList.MouseMove
			Dim tree As TreeList = TryCast(sender, TreeList)
			Dim info As TreeListHitInfo = tree.CalcHitInfo(e.Location)
			If info.HitInfoType = HitInfoType.NodeCheckBox Then
				HotTrackCheckBoxRectangle = info.Bounds
			Else
				HotTrackCheckBoxRectangle = Rectangle.Empty
			End If
		End Sub
	End Class
End Namespace
